### PR TITLE
Certification tests for HashiCorp Vault Secret Store.

### PR DESCRIPTION
**Pull request by tmacam**
# Description

This PR adds certification tests for HashiCorp Vault Secret Store.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1926
## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by artursouza** [comment]
Why do we need to wait for 1.9 to add the commented out code?

**Comment by tmacam** [comment]
@artursouza

> Why do we need to wait for 1.9 to add the commented out code?

Due to the problem that https://github.com/dapr/dapr/pull/5208 addresses. When we run locally the tests they are not picking a version of dapr with that fix.

File: tests/certification/secretstores/hashicorp/vault/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+## Basic Test for CRUD operations:
+1. Able to create and test connection.
+2. Able to do retrieve secrets.
+3. Negative test to fetch record with key, that is not present.
+
+## Test network instability
+1. Vault component does not expose a time out configuration option. For this test, let's assume a 1 minute timeout.
+2. Retrieve a key to show the connection is fine.
+3. Interrupt the network (Vault port, 8200git pus) for longer than the established timeout value.
```
Comment (artursouza) [inline tests/certification/secretstores/hashicorp/vault/README.md:19]: Typo?

Comment (tmacam) [reply tests/certification/secretstores/hashicorp/vault/README.md:19]: Yeah! :( Fixing...

**Comment by berndverst** [comment]
Pin the appropriate version of dapr/dapr (a specific commit) in certification tests to avoid waiting for 1.9 release. No need to wait.

File: tests/certification/secretstores/hashicorp/vault/vault_test.go [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+		Step(sidecar.Run(sidecarName,
+			embedded.WithoutApp(),
+			embedded.WithComponentsPath(secretStoreComponentPath),
+			embedded.WithDaprGRPCPort(currentGrpcPort),
+			embedded.WithDaprHTTPPort(currentHttpPort),
+			componentRuntimeOptions(),
+		)).
+		Step("Waiting for component to load...", flow.Sleep(5*time.Second)).
+		Step("Verify component is registered", testComponentFound(t, secretStoreName, currentGrpcPort)).
+		Step("Verify component has support for multiple key-values under the same secret",
```
Comment (tmacam) [inline tests/certification/secretstores/hashicorp/vault/vault_test.go:153]: FYI: the functions referenced here could be used in other tests as-is. And TBH, checking that the component  we _think_ we registered is indeed loaded is a good sanity check in those tests.

**Comment by tmacam** [comment]
> Pin the appropriate version of dapr/dapr (a specific commit) in certification tests to avoid waiting for 1.9 release. No need to wait.

@berndverst  done

**Review by artursouza — APPROVED** [review]

**Comment by berndverst** [comment]
@artursouza you cannot merge this without running `/ok-to-test`...

**Comment by berndverst** [comment]
Revert this if these tests fail: https://github.com/dapr/components-contrib/actions/runs/3153995296

**Comment by tmacam** [comment]
> Revert this if these tests fail: https://github.com/dapr/components-contrib/actions/runs/3153995296

It passed all tests ✅

**Pull request merged**